proposals/csharp-10.0/parameterless-struct-constructors.md

```diff
@@ -0,0 +1,157 @@
+# Parameterless struct constructors
+
+## Summary
+
+Support parameterless constructors and instance field initializers for struct types.
+
+## Proposal
+
+### Instance field initializers
+Instance field declarations for a struct may include initializers.
+
+As with [class field initializers](https://github.com/dotnet/csharplang/blob/master/spec/classes.md#instance-field-initialization):
+> A variable initializer for an instance field cannot reference the instance being created. 
+
+### Constructors
+A struct may declare a parameterless instance constructor.
+The struct can be declared as `struct`, `readonly struct`, or `ref struct`.
+
+If the struct does not declare a parameterless instance constructor, and the struct has no fields with variable initializers, the struct (see [struct constructors](https://github.com/dotnet/csharplang/blob/master/spec/structs.md#constructors)) ...
+> implicitly has a parameterless instance constructor which always returns the value that results from setting all value type fields to their default value and all reference type fields to null.
+
+If the struct does not declare a parameterless instance constructor, and the struct has field initializers, a `public` parameterless instance constructor is synthesized.
+The parameterless constructor is synthesized even if all initializer values are zeros.
+
+### Modifiers
+A parameterless instance constructor may be less accessible than the struct.
+
+A parameterless instance constructor may be declared `extern`.
```

Comment:
Can they be `partial`? 

I think we should start out with a base statement here, essentially something along the lines of 

> A parameterless struct constructor can have the same set of modifiers as other struct instance constructors. The only exceptions are ... 

Assuming there are any exceptions (can't think of any right now).

Reply:
Added a comment that the same set of modifiers apply to parameterless constructors. Constructors cannot be declared `partial` though.

---
In reply to: [583304832](https://github.com/dotnet/csharplang/pull/4459#discussion_r583304832) [](ancestors = 583304832)